File: python3/2016.py
class Solution:
    def maximumDifference(self, n: List[int]) -> int:
        p = n[0]
        m = -1
        for i in range(len(n)):
            if p < n[i]:
                m = max(m, n[i] - p)
            else:
                p = n[i]
        return m

File: dart/2016.dart
class Solution {
    int max(int x, int y){
        return (x > y) ? x : y;
    }

    int maximumDifference(List<int> n) {
        int p = n[0], l = n.length, m = -1;
        for (int i = 1; i < l; i++)
            if (p < n[i])
                m = max(m, n[i] - p);
            else
                p = n[i];
        return m;
   }
}
